python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_decorators.py

```diff
@@ -443,6 +444,34 @@ def my_asset(_input1: str):
         pass
 
     assert my_asset.op.input_defs[0].dagster_type.display_name == "String"
+    assert my_asset.op.input_defs[0].dagster_type.typing_type == str
+
+
```

Comment:
[1] we should add some tests to clarify the choice, something like
```
@op
def produce() -> None:
    pass


@op
def consume(x):
    return x


@graph
def ok_or_invalid():
    consume(produce())
```

and the equivalent in `@asset` - assuming that `-> None` becoming `Nothing` would cause you to change how your asset deps are defined.